supertokens_python/recipe/emailpassword/emaildelivery/service/backwardCompatibility/__init__.py

```diff
@@ -0,0 +1,104 @@
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, Awaitable, Callable, Dict, Union, cast
+
+from supertokens_python.ingredients.emaildelivery.types import \
+    EmailDeliveryInterface
+from supertokens_python.recipe.emailpassword.interfaces import \
+    TypeEmailPasswordEmailDeliveryInput
+from supertokens_python.recipe.emailpassword.recipe import EmailPasswordRecipe
+from supertokens_python.recipe.emailpassword.types import User
+from supertokens_python.recipe.emailpassword.utils import (
+    InputEmailVerificationConfig, InputResetPasswordUsingTokenFeature,
+    default_create_and_send_custom_email)
+from supertokens_python.recipe.emailverification.emaildelivery.services.backwardCompatibility import \
+    BackwardCompatibilityService as \
+    EmailVerificationBackwardCompatibilityService
+from supertokens_python.recipe.emailverification.interfaces import \
+    TypeEmailVerificationEmailDeliveryInput
+from supertokens_python.recipe.emailverification.types import \
+    User as EmailVerificationUser
+from supertokens_python.supertokens import AppInfo
+
+
+class CreateAndSendCustomEmailInput:
+    user: str
+    email_verification_url_with_token: str
+    user_context: Dict[str, Any]
+
+class BackwardCompatibilityService(EmailDeliveryInterface[TypeEmailPasswordEmailDeliveryInput]):
+    app_info: AppInfo
+    emailVerificationBackwardCompatiblityService: EmailVerificationBackwardCompatibilityService
+
+    def __init__(self,
+            app_info: AppInfo,
+            recipeInterfaceImpl: EmailPasswordRecipe, # Any,  # FIXME: Should be EmailPasswordRecipe. But leads to circular dependency
+            reset_password_using_token_feature: Union[InputResetPasswordUsingTokenFeature, None] = None,
+            email_verification_feature: Union[InputEmailVerificationConfig, None] = None,
+        ) -> None:
+        self.app_info = app_info
+        self.recipeInterfaceImpl = recipeInterfaceImpl
+        self.reset_password_using_token_feature = reset_password_using_token_feature
+
+        if reset_password_using_token_feature:
+            if reset_password_using_token_feature.create_and_send_custom_email is None:
+                reset_password_using_token_feature.create_and_send_custom_email = default_create_and_send_custom_email(self.app_info)
+            self.reset_password_using_token_feature = reset_password_using_token_feature
+
+        if email_verification_feature:
+            if email_verification_feature.create_and_send_custom_email is not None:
+                async def create_and_send_custom_email_wrapper(user: User, link: str, user_context: Dict[str, Any]):
+                    user_info = await self.recipeInterfaceImpl.recipe_implementation.get_user_by_id(user.user_id, user_context)
+                    if user_info is None:
+                        raise Exception("Unknown User ID provided")
+
```

Comment:
you can do this instead of assert and cast
```python
create_and_send_custom_email: Union[Callable[[
                EmailVerificationUser, str, Dict[str, Any]], Awaitable[None]], None] = None
if email_verification_feature:
    if email_verification_feature.create_and_send_custom_email is not None:
        tmp = email_verification_feature.create_and_send_custom_email
        async def create_and_send_custom_email_wrapper(user: EmailVerificationUser, link: str, user_context: Dict[str, Any]):
            user_info = await self.recipeInterfaceImpl.recipe_implementation.get_user_by_id(user.user_id, user_context)
            if user_info is None:
                raise Exception("Unknown User ID provided")

            return await tmp(user_info, link, user_context)

        create_and_send_custom_email = create_and_send_custom_email_wrapper

self.emailVerificationBackwardCompatiblityService = EmailVerificationBackwardCompatibilityService(
    app_info, create_and_send_custom_email=create_and_send_custom_email
)
```